packages/mutation-testing-metrics/src/model/test-model.ts

```diff
@@ -0,0 +1,32 @@
+import { OpenEndLocation, TestDefinition } from 'mutation-testing-report-schema';
+import { MutantModel } from './mutant-model';
+
+export class TestModel implements TestDefinition {
+  id!: string;
+  name!: string;
+  location?: OpenEndLocation | undefined;
+
+  killedMutants?: MutantModel[];
+  coveredMutants?: MutantModel[];
+
+  public addCovered(mutant: MutantModel) {
+    if (!this.coveredMutants) {
+      this.coveredMutants = [];
+    }
+    this.coveredMutants.push(mutant);
+  }
+
+  public addKilled(mutant: MutantModel) {
+    if (!this.killedMutants) {
+      this.killedMutants = [];
+    }
+    this.killedMutants.push(mutant);
+  }
+
+  constructor(input: TestDefinition) {
+    Object.entries(input).forEach(([key, value]) => {
+      // @ts-expect-error dynamic assignment so we won't forget to add new properties
```

Comment:
Won't we still have to any new properties to this class if they are added? I'm unsure what extra work this is preventing

Reply:
I'm using this pattern 4 times in total. Each time it saves me from copying over some properties and making accidental copy-and-paste errors. 🤷‍♂️

Reply:
Fair enough